ProjFS.Mac/PrjFSKext/PrjFSKext/KauthHandler.cpp

```diff
@@ -256,12 +254,26 @@ static int HandleVnodeOperation(
     // arg2 is the (vnode_t) parent vnode
     int* kauthError =       reinterpret_cast<int*>(arg3);
     int kauthResult = KAUTH_RESULT_DEFER;
+    bool putVnodeWhenDone = false;
+
+    // A lot of our file checks such as attribute tests behave oddly if the vnode
+    // refers to a named fork/stream; apply the logic as if the vnode operation was
+    // occurring on the file itself. (/path/to/file/..namedfork/rsrc)
+    if (vnode_isnamedstream(currentVnode))
+    {
+        vnode_t mainFileFork = vnode_getparent(currentVnode);
```

Comment:
What should we do if this call fails and returns `NULLVP`? We now know that `currentVnode` is referring to a named stream and that we failed to resolve to the actual file, so should we give up?

Reply:
Related: is there any way to guarantee that we can find the parent? The `vnode_getparent` function is racy since it only resolves vnodes that are still in the name cache. Is there an alternative?

Reply:
I'm happy to leave the bigger investigation for a separate change, but I think we should either assert or return a failure here if this call fails. Otherwise we know we will misbehave downstream of this code.

Reply:
OK, an assertion sounds the most sensible approach here.